### ISSUE TITLE
Show the modal on submission rejection

### ISSUE THREAD

**Issue by NabbeunNabi** [ourjapanlife/findadoc-web#770]
# Problem
Rejecting a submission is a big process. So we want to be sure before rejecting the facility. Right now the modal only displays on the `ModEditSubmissionForm` for accepting a facility and the back button. We also want it to show for rejection.

# Success
The modal shows with correct messaging for clicking on the rejection button. Make sure there are tests for this in `moderation.cy.ts`
### Requirements

- [x] Implement modal display on rejection button press.
- [x] Have tests for reject back button display

**Comment by NabbeunNabi** [comment]
This will be ready once this PR is merged https://github.com/ourjapanlife/findadoc-web/pull/769

**Comment by theyokohamalife** [comment]
@NabbeunNabi For the requirements, what do we mean by tests for each back button display? We just need tests for the rejection behavior, right?

**Comment by NabbeunNabi** [comment]
@theyokohamalife 

Yes...but maybe I was trying to be a bit sneaky by saying in this they could add the tests for the others as well since those were not implemented in the previous ones that implemented this behavior.

Should I pull the tests to a separate ticket?

**Comment by theyokohamalife** [comment]
@NabbeunNabi Yeah, I think we should create a separate ticket for unrelated tests.

**Comment by theyokohamalife** [comment]
@NabbeunNabi This ticket is bit confusing, because the title says to show the modal on submission rejection. However, the success and requirements details add additional features. At first glance I thought this ticket was for "Show modal when user clicks reject button."

**Comment by NabbeunNabi** [comment]
@theyokohamalife I updated it so the words only focused on the reject button
